### ISSUE TITLE
PEP 3105 Make print a function in twisted/test

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8397]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8397|
|Type|enhancement|
|Created|2016-06-03 10:47:20Z|

This ticket was split out of [ticket:5812] which was too big to review.

PEP 3105 ( ​​​https://www.python.org/dev/peps/pep-3105/ ) **print** deprecates has changed from a statement to a function.

In Python 2.6 and Python 2.7, the default is print as a statement

```
print "Hello"
```

In Python 3.x, the above example is an error, since only print as a function is available.

To consistently use **print** as a function
on Python 2.6, Python 2.7, and Python 3.x, this should be done:

```
from __future__ import print_function

print("Hello")
```

<details><summary>Searchable metadata</summary>

```
trac-id__8397 8397
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464950840271489 1464950840271489
changetime__1466752689501903 1466752689501903
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/113

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli set status to `closed`|
|-|-|

I mentioned in the PR that this module doesn't actually have an print statements that are uncommented, so we can probably avoid importing print_function in the module until there is a use case for it.

Thank you for all your work with all the other modules, though, rodrigc!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set status to `reopened`|
|-|-|

See my feedback in https://github.com/twisted/twisted/pull/113#issuecomment-224804742

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @rodrigc<br>@adiroiban set status to `new`|
|-|-|

Changes look good.
Please create a branch in twisted, send it to buildbot and merge after buildbot is green (exceptiong twistedchecker).

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset b93dd8d984f4684dd536792eea5e9ad113439a99
```
#!CommitTicketReference repository="" revision="b93dd8d984f4684dd536792eea5e9ad113439a99"
Merge 8397-rodrigc-test: Convert from print statement to print() function

Author: rodrigc
Reviewer: cyli, alex, adiroiban
Fixes: #8397
```
